### ISSUE TITLE
Caching @import if the file names similar

### ISSUE THREAD

**Issue by tavriaforever** [stylus/stylus#1951]
stylus@0.51.1

## What the problem?
Founded a bug when stylus caches disclosure @import, as a result, if the paths have similar parts – styles taken result css from the first import as many times as imports.

### For example:
Two `styl` file
**path 1**: `../../libs/blocks/icon/icon.styl`
**content**:
```stylus
.icon
{
    display: inline-block;

    text-align: center;

    background: 50% no-repeat;

    /* Hack for correct baseline positioning */
    &:empty:after
    {
        visibility: hidden;

        content: '\00A0';
    }

    /*
     * Чтобы иконка правильно позиционировалась внутри блочного контекста,
     * нужно прописать родителю свойство line-height со значением, равным высоте иконки
     */
    & > img,
    & > svg
    {
        margin: -5.15em 0 -5em; /* 0.15 — magic number, empirically found */

        vertical-align: middle;
    }
}

```

**path 2**: `../../blocks/icon/icon.styl`
**content**:
```stylus
.icon
{
    width: 200px;
    height: 200px;
    color: #000;
}
```

Index.styl file contain two imports:
```stylus
@import "../../libs/blocks/icon/icon.styl";
@import "../../blocks/icon/icon.styl";

```

### Result:
```
.icon {
  display: inline-block;
  text-align: center;
  background: 50% no-repeat;
/* Hack for correct baseline positioning */
/*
     * Чтобы иконка правильно позиционировалась внутри блочного контекста,
     * нужно прописать родителю свойство line-height со значением, равным высоте иконки
     */
}
.icon:empty:after {
  visibility: hidden;
  content: '\00A0';
}
.icon > img,
.icon > svg {
  margin: -5.15em 0 -5em; /* 0.15 — magic number, empirically found */
  vertical-align: middle;
}
.icon {
  display: inline-block;
  text-align: center;
  background: 50% no-repeat;
/* Hack for correct baseline positioning */
/*
     * Чтобы иконка правильно позиционировалась внутри блочного контекста,
     * нужно прописать родителю свойство line-height со значением, равным высоте иконки
     */
}
.icon:empty:after {
  visibility: hidden;
  content: '\00A0';
}
.icon > img,
.icon > svg {
  margin: -5.15em 0 -5em; /* 0.15 — magic number, empirically found */
  vertical-align: middle;
}
```

## Important!
This bug depends on option `resolve url`. 
https://github.com/stylus/stylus/blob/48569e1427696447c7d7aba78f87c5ece1ddec49/test/run.js#L27
Without setting this option, the renderer works fine.

## How to reproduced bug?
Clone https://github.com/tavriaforever/stylus-project/tree/stylus-%40import-bug

``` bash
npm install
npm run build
```

// followers: @blond @tadatuta @sipayrt

**Comment by blond** [comment]
+1

**Comment by sipayRT** [comment]
+1

**Comment by SwinX** [comment]
+1

**Comment by unlok** [comment]
+1

**Comment by alexbaumgertner** [comment]
+1

**Comment by Panya** [comment]
Can you try this with `new-resolver` branch (`stylus/stylus#new-resolver`)? It should work fine.

**Comment by tavriaforever** [comment]
`new-resolver` works great! :guitar:
Are you planning to release the stable `stylus` version with the new resolver?

**Comment by Panya** [comment]
Yep, we will release it later on this week.

**Comment by tavriaforever** [comment]
Thank you very much! Let's wait for the release impatiently.

**Comment by Panya** [comment]
New resolver is available in `0.52.0` with `nocheck: true` option (or `--resolve-url-nocheck` CLI flag):
```js
stylus.define('url', stylus.resolver({ nocheck: true });
```

**Comment by tavriaforever** [comment]
Thanks a lot!
